### PR TITLE
Removing the make_module script.

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24151 Replacing axis with dim in quantized cat
* **#23635 Removing the make_module script.**

n128675

Differential Revision: [D16593364](https://our.internmc.facebook.com/intern/diff/D16593364)

### CODE REVIEW

File: torch/nn/modules/_wrapped_modules.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from __future__ import unicode_literals
+
+import torch
+from torch.nn.modules import Module
+
+r"""Base class for all wrapper modules."""
+class _BaseWrapperModule(Module):
+    def __init__(self):
+        super(_BaseWrapperModule, self).__init__()
+
```
Comment (jerryzh168) [inline torch/nn/modules/_wrapped_modules.py]: should we do torch.operation = None or something

File: torch/nn/modules/_wrapped_modules.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    def forward(self, *args):
+        return torch.operation(*args)
```
Comment (jerryzh168) [inline torch/nn/modules/_wrapped_modules.py]: self.operation?

File: torch/nn/quantized/modules/_wrapped_modules.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        self.register_buffer('scale', torch.tensor([1.0], dtype=torch.double))
+        self.register_buffer('zero_point', torch.tensor([0], dtype=torch.long))
+        self.operation = None
+
+    def forward(self, *args):
+        return self.operation(*args, scale=self.scale,
+                              zero_point=self.zero_point)
+
+    @classmethod
+    def from_float(cls, mod):
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply torch/nn/quantized/modules/_wrapped_modules.py]: e.g. https://github.com/pytorch/pytorch/blob/master/torch/nn/qat/modules/conv.py#L40

File: torch/nn/quantized/modules/_wrapped_modules.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        scale, zero_point = mod.observer.calculate_qparams()[:2]
+        mod = cls()
+        mod.scale = torch.tensor(scale, dtype=torch.double)
+        mod.zero_point = torch.tensor(zero_point, dtype=torch.long)
+        return mod
+
+
+r"""Add module wraps torch.ops.quantized.add."""
+class Add(_BaseWrapperModule):
+    __FLOAT_MODULE = torch.nn.modules.Add
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/_wrapped_modules.py]: nit: torch.nn.Add?

File: torch/nn/quantized/modules/_wrapped_modules.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import torch
+from torch.nn.modules import Module
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/_wrapped_modules.py]: nit: torch.nn import Module?

File: torch/nn/quantized/modules/_wrapped_modules.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (zafartahirov) [reply torch/nn/quantized/modules/_wrapped_modules.py]: If qconfig is provided, should the scale and zero_point come from `qconfig`?

File: torch/nn/quantized/modules/_wrapped_modules.py (hunk 5 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/_wrapped_modules.py]: I think, that potentially can cause circular deps -- lemme check if I can

File: torch/nn/modules/_wrapped_modules.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import torch
+from torch.nn.modules import Module
+
+r"""Base class for all wrapper modules."""
+class _BaseWrapperModule(Module):
```
Comment (dzhulgakov) [inline torch/nn/modules/_wrapped_modules.py]: this class is likely not going to be scriptable - plesae double-check

**Review by dzhulgakov — CHANGES_REQUESTED** [review]
Unittests that the class is scriptable? (the whole point of getting rid of magic)

File: torch/nn/quantized/modules/_wrapped_modules.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jerryzh168) [reply torch/nn/quantized/modules/_wrapped_modules.py]: yes

File: torch/nn/modules/_wrapped_modules.py (hunk 6 above)
Comment (jerryzh168) [reply torch/nn/modules/_wrapped_modules.py]: actually I think we probably don't need them to be scriptable since these are not used in graph mode, or are you referring to the case when we quantize in eager mode and then script it?

File: torch/nn/quantized/modules/_wrapped_modules.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jerryzh168) [reply torch/nn/quantized/modules/_wrapped_modules.py]: actually, this doesn't make sense for activation, what we want to pass in is not qconfig, but a 'quantizer' which doesn't exist yet. we can do this later.

File: torch/nn/modules/_wrapped_modules.py (hunk 6 above)
Comment (dzhulgakov) [reply torch/nn/modules/_wrapped_modules.py]: Yes, unless there's a good reason not to, our default should be that everything is scriptable.

In this case, it's a very likely case that I want to quantize using eager mode, tweak my model and script the result

File: torch/nn/modules/__init__.py [hunk 7]
```diff
@@ -53,10 +55,7 @@
     'ConstantPad3d', 'Bilinear', 'CosineSimilarity', 'Unfold', 'Fold',
     'AdaptiveLogSoftmaxWithLoss', 'TransformerEncoder', 'TransformerDecoder',
     'TransformerEncoderLayer', 'TransformerDecoderLayer', 'Transformer',
-    'Flatten'
+    'Flatten',
+    # Wrapped modules
+    'Add'
```
Comment (soumith) [inline torch/nn/modules/__init__.py]: why are you adding an `nn.Add`?

**Review by soumith — CHANGES_REQUESTED** [review]

**Comment by dzhulgakov** [comment]
Btw, this PR could have benefited from:
- better description that provides a bit of context (e.g. copy-paste from quantization design doc)
- unittest to verify that the module works and is scriptable (at least one of them)

File: torch/nn/quantized/modules/wrapper_module.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    torch.add: ops.quantized.add
+}
+
+class WrapperModule(torch.nn.Module):
+    def __init__(self, operation):
+        super(WrapperModule, self).__init__()
+        self.operation = operation
+
+    def forward(self, x):
+        return self.operation(*x)
```
Comment (dzhulgakov) [inline torch/nn/quantized/modules/wrapper_module.py]: cool, didn't know our JIT is smart enough to resolve `self.operation` (original reason to ask for explicit module)

why are you unpacking the args though? it's confusing to user and doesn't work with JIT (as it doesn't support varargs).

I guess we can just hard-code 2 arguments then or have per-num-arguments versions.

**Review by dzhulgakov — CHANGES_REQUESTED** [review]
if you claim it's scriptable - where's the unittest to verify it's so?

File: torch/nn/quantized/modules/wrapper_module.py (hunk 8 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/wrapper_module.py]: I am not 100% sure if we will support unary ops through this mechanism. This is more of a "generic" implementation.

Comment (zafartahirov) [reply torch/nn/quantized/modules/wrapper_module.py]: I think it works with variable length, but only when using `torch.jit.trace`

Comment (dzhulgakov) [reply torch/nn/quantized/modules/wrapper_module.py]: I'd rather make it work with Script too. You can create a two-argument version for now and hide it under `make_wrapper` function. In future, it'd be possible to extend to more arguments if needed (by inspecting schema of the passed operation)

Comment (zafartahirov) [reply torch/nn/quantized/modules/wrapper_module.py]: We don't have the `make_wrapper` anymore -- it's only a wrapper class :/

Comment (dzhulgakov) [reply torch/nn/quantized/modules/wrapper_module.py]: again, I think having them scriptable is important - and starting with two-argument ops is fine. You could introduce `make_wrapper` to allow future insertion point for other number of arguments

**Review by dzhulgakov — CHANGES_REQUESTED** [review]

File: torch/nn/quantized/modules/wrapper_module.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+
+"""Wrappes unary operators."""
+class UnaryWrapper(Module):
+    def __init__(self, operation):
+        super(UnaryWrapper, self).__init__()
+        self.operation = operation
+
+    def forward(self, x):
+        return self.operation(x)
```
Comment (raghuramank100) [inline torch/nn/quantized/modules/wrapper_module.py]: Need to have type annotation for this to work
```
class UnaryWrapper(nn.Module):
    def __init__(self, operation):
        super(UnaryWrapper, self).__init__()
        self.operation = operation

    def forward(self, x):
        # type: (List[Tensor]) -> Tensor
        return self.operation(x)
```

File: torch/nn/quantized/modules/wrapper_module.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+"""Wraps unary operators (quantized)."""
+class QuantizedUnaryWrapper(Module):
+    def __init__(self, operation):
+        super(WrapperModule, self).__init__()
+        self.operation = operation
+        self.register_buffer('scale', torch.tensor([1.0], dtype=torch.double))
+        self.register_buffer('zero_point', torch.tensor([0], dtype=torch.long))
+
+    def forward(self, x):
```
Comment (raghuramank100) [inline torch/nn/quantized/modules/wrapper_module.py]: Need type annotation

File: test/test_quantization.py [hunk 11]
```diff
@@ -290,6 +290,72 @@ def checkQuantized(model):
...
+        y_hat = script_module((a, b))
+        self.assertEqual(y, y_hat,
+                         "Results of the traced unary wrapper are off!")
+        y_hat = script_qmodule((a, b))
+        self.assertEqual(y, y_hat,
+                         "Results of the traced unary qwrapper are off!")
+
+        BinaryWrapper = nnq.modules.wrapper_module.BinaryWrapper
+        QBinaryWrapper = nnq.modules.wrapper_module.QuantizedBinaryWrapper
+        script_module = torch.jit.trace(BinaryWrapper(torch.add),
```
Comment (raghuramank100) [inline test/test_quantization.py]: Need to have check_trace=False for tracing to work. I have filed a github issue () to track this bug.
https://github.com/pytorch/pytorch/issues/23993

File: test/test_quantization.py [hunk 12]
```diff
@@ -290,6 +290,72 @@ def checkQuantized(model):
         checkQuantized(model)
 
 
+class ScriptabilityTest(QuantizationTestCase):
+    def test_wrapper_traceable(self):
+        UnaryWrapper = nnq.modules.wrapper_module.UnaryWrapper
+        QUnaryWrapper = nnq.modules.wrapper_module.QuantizedUnaryWrapper
+        script_module = torch.jit.trace(UnaryWrapper(torch.cat),
+                                        (torch.tensor(0), torch.tensor(0)))
+        script_qmodule = torch.jit.trace(QUnaryWrapper(torch.cat),
```
Comment (raghuramank100) [inline test/test_quantization.py]: These tests will not work for tracing or scripting. This is because the operators can take dynamic length inputs, one can trace with a tuple of length two and then run with a tuple of different length. A test case that corresponds to actual use is:

An notebook with the outputs is at: https://our.internmc.facebook.com/intern/anp/view/?id=128232
```python
import torch.nn as nn

class UnaryWrapper(nn.Module):
    def __init__(self, operation):
        super(UnaryWrapper, self).__init__()
        self.operation = operation

    def forward(self, x):
        # type: (List[Tensor]) -> Tensor
        return self.operation(x)


class BinaryWrapper(nn.Module):
    def __init__(self, operation):
        super(BinaryWrapper, self).__init__()
        self.operation = operation

    def forward(self, x,y):
        return self.operation(x,y)


class testModel(nn.Module):
    def __init__(self):
        super(testModel, self).__init__()
        self.mycat = UnaryWrapper(torch.cat)
        self.myadd= BinaryWrapper(torch.add)
        self.mymul = BinaryWrapper(torch.mul)
    
    def forward(self,x):
        y = self.mycat([x ,x ,x])
        z = self.myadd(y, y)
        z = self.mymul(z,z)
        return z
  ```

**Review by raghuramank100 — CHANGES_REQUESTED** [review]
Based on discussions with @jamesr66a have suggested changes that are required for tracing and scripting to work.

File: torch/nn/quantized/modules/wrapper_module.py (hunk 9 above)
Comment (dzhulgakov) [reply torch/nn/quantized/modules/wrapper_module.py]: This annotation is only valid for Cat (which takes the list). Other unary ops, shall we choose to quantize them, should have `Tensor` annotation.

I really think it's should be an implementation details. We can expose a single function `make_wrapper(torch.something)` and have internal dictionary dispatch to pick the right wrapper (out of a few - like binary, unary, list). All these wrappers aren't that generic anyway since _FLOAT_MODULES explicitly lists supported functions.

**Review by dzhulgakov — CHANGES_REQUESTED** [review]

File: test/test_quantization.py [hunk 13]
```diff
@@ -290,6 +292,34 @@ def checkQuantized(model):
         checkQuantized(model)
 
 
+class ScriptabilityTest(QuantizationTestCase):
+    def test_wrapper_traceable(self):
+        model_under_test = ModForWrapping()
+        x = torch.rand(10)
```
Comment (raghuramank100) [inline test/test_quantization.py]: Can you combine test_wrapper_traceable and test_wrapper_scriptable?

File: torch/quantization/quantize.py [hunk 14]
```diff
@@ -70,7 +70,10 @@ def add_observer(module, skip_list=DEFAULT_SKIP_LIST):
             forward_hooks
     """
     for child in module.children():
-        add_observer(child)
+        if type(child) == nnq.FloatFunctional:
```
Comment (raghuramank100) [inline torch/quantization/quantize.py]: This code should be called once you know you are at a leaf module and can check for qconfig:
Remove this change here and insert below:
```python
 # Insert observers only for leaf nodes, note that this observer is for
    # the output of the module, for input QuantStub will observe them
    if hasattr(module, 'qconfig') and module.qconfig is not None and \
       len(module._modules) == 0 and type(module) not in skip_list:
        # observer and hook will be gone after we swap the module
        if type(module) == nnq.FloatFunctional:
              module.observer = module.qconfig.activation()
      else:
           module.add_module('observer', module.qconfig.activation())
           module.register_forward_hook(_observer_forward_hook)
```

Comment (zafartahirov) [reply torch/quantization/quantize.py]: This is already there later in the code. At the same time, FloatFunctional is not expected to have children, so it will always be a child. The reason is that the FloatFunctional will raise an error if `forward` is called, which means it's always a leaf. At the same time, because this this is a recursive function, semantically it is the same

Comment (zafartahirov) [reply torch/quantization/quantize.py]: Made the change anyway :)

**Review by raghuramank100 — APPROVED** [review]
Approved, please address comments before landing.

File: aten/src/ATen/native/quantized/cpu/qconcat.cpp [hunk 15]
```diff
@@ -22,7 +22,7 @@ bool is_valid_quantization_scheme(Tensor t) {
 template <bool ReLUFused>
 Tensor quantized_cat(
     const std::vector<Tensor>& qxs,
-    int64_t axis,
+    int64_t dim,
```
Comment (dzhulgakov) [inline aten/src/ATen/native/quantized/cpu/qconcat.cpp]: it's probably fine now, but in general - don't mix unrelated changes in the same PR

File: torch/nn/quantized/modules/functional_modules.py [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+    def forward(self, x):
+        raise RuntimeError("FloatFunctional is not intended to use the " +
+                           "'forward'. Please use the underlying operation")
+
+    def add(self, x, y):
+        # type: (Tensor, Tensor) -> Tensor
+        assert self.op_name == 'add', \
+            "Running add while initialized as {}".format(self.op_name)
+        r = torch.add(x, y)
+        self.observer(r)
```
Comment (dzhulgakov) [inline torch/nn/quantized/modules/functional_modules.py]: this might not work with QAT, because I think observer returns a value override in that case - cc @jerryzh168 . Maybe put a TODO now and address it in a separate diff?

File: torch/nn/quantized/modules/functional_modules.py [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+        return r
+
+    def cat(self, x, dim=None):
+        # type: (List[Tensor], Optional[int]) -> Tensor
+        assert self.op_name == 'cat', \
+            "Running cat while initialized as {}".format(self.op_name)
+        if dim is None:
```
Comment (dzhulgakov) [inline torch/nn/quantized/modules/functional_modules.py]: add a TODO to fix op registration. This logic here feels out of place

File: torch/nn/quantized/modules/functional_modules.py [hunk 18]
```diff
@@ -0,0 +1,76 @@
+import torch
+from torch._ops import ops
+
+
+_VALID_OP_NAMES = {
+    'add',
+    'cat'
+}
+
+class FloatFunctional(torch.nn.Module):
```
Comment (dzhulgakov) [inline torch/nn/quantized/modules/functional_modules.py]: comments on this class? can be something simple explaining that it's a substitute for torch.*

File: torch/nn/quantized/modules/functional_modules.py [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+    def __init__(self, op_name):
```
Comment (dzhulgakov) [inline torch/nn/quantized/modules/functional_modules.py]: why do you need to initialize with op_name? I think it's much nicer API if you don't have to specify it - then your __init__ would be very simple. You could add some handling inside to ensure that only one method is ever called, but I'd say even that is optional.

**Review by dzhulgakov — CHANGES_REQUESTED** [review]

File: torch/nn/quantized/modules/functional_modules.py (hunk 19 above)
Comment (raghuramank100) [reply torch/nn/quantized/modules/functional_modules.py]: I had suggested this to protect from users doing
```
 # in init
self. f1 = nn.quantize.functionalwrapper()

def forward(self ,x, y):
   self.f1.add(x,y)
   self.f1.cat([x,x])
```
This should be an error: We cannot have f1 be used for multiple operations, as the observers have to be per instance.

Also, explicitly specifying the operation at init is consistent with how modules are used. A module does a fixed operation, defined at init. With the wrapper, it is hard to determine what the operations are looking at the init.

File: torch/nn/quantized/modules/functional_modules.py (hunk 16 above)
Comment (raghuramank100) [reply torch/nn/quantized/modules/functional_modules.py]: Good catch!, we will need to return a value for QAT.  @jerryzh168 : Even the existing code has a problem: We are calling fake quant in the forward hook and this needs to change numerics, which is not allowed.

Comment (zafartahirov) [reply torch/nn/quantized/modules/functional_modules.py]: So should that be `r = self.observer(r)`?

File: aten/src/ATen/native/quantized/cpu/qconcat.cpp (hunk 15 above)
Comment (zafartahirov) [reply aten/src/ATen/native/quantized/cpu/qconcat.cpp]: My bad -- this was bothering me a lot, and I was too lazy to have a stack :) Will not do that again!

File: torch/nn/quantized/modules/functional_modules.py (hunk 17 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/functional_modules.py]: The registtration will support optional arguments. The thing is that I still believe we should resolve it here, instead of doing it  inside of the kernel. The reason is that Once the documentation is parsed, the default values will be shown. But if we do it in the kernel, the default value will be hidden from the user.

File: torch/nn/quantized/modules/functional_modules.py (hunk 19 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/functional_modules.py]: I am not sure what to do here -- we had a long discussion about that, so we need to make a decision soon if we need that or not.

File: torch/nn/quantized/modules/functional_modules.py (hunk 17 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/functional_modules.py]: I can make it optional with default value of 0

Comment (dzhulgakov) [reply torch/nn/quantized/modules/functional_modules.py]: If it's a bug - then file an issue against JIT/op bindings to fix it :) It feels out of place here.

File: torch/nn/quantized/modules/functional_modules.py (hunk 16 above)
Comment (dzhulgakov) [reply torch/nn/quantized/modules/functional_modules.py]: QAT observers return overrides

File: torch/nn/quantized/modules/functional_modules.py (hunk 19 above)
Comment (dzhulgakov) [reply torch/nn/quantized/modules/functional_modules.py]: I'd vote for not specifying it in constructor and @soumith was voting for the same. It's strictly more lines of code and still not a complete protection as I can mix up two different adds together. The benefit of not specifying it in constructor is that one can build a nice wrapper like (for eager only, but still cute):

```
def __init__(...):
  self.stats = [nn.quantize.functionalwrapper() for i in range(10)]

def forward():
  counter = Counter()
  z = self.stats[counter.next()].add(x,y)
  return self.stats[counter.next()].cat([z,z])
```

We shouldn't provide it as default, but it feels more flexible.

We could remember the first op being invoked in a field on python side and re-check it next time. But it'd be interfere with torchscript a bit today.

Comment (raghuramank100) [reply torch/nn/quantized/modules/functional_modules.py]: Agree with the comment about incomplete protection, but isn't some protection better than none?.  But do not understand the benefit of the wrapper. One could always do:
```
def __init__(...):
  opList = ['add', 'cat', ....]
  self.stats = [nn.quantize.functionalwrapper(opList[i]) for i in range(len(opList)]

def forward():
  counter = Counter()
  z = self.stats[counter.next()].add(x,y)
  return self.stats[counter.next()].cat([z,z])
```
@dzhulgakov : If you are convinced about having no argument in the constructor, lets go with it.

Comment (zafartahirov) [reply torch/nn/quantized/modules/functional_modules.py]: I am ok with either implementation. However, to clarify the purpose of the name in the constructor is to make sure that the functionalwrapper is used as intended in the constructor. I can see benefits of it from usage safety. But at the same time, I think this is getting into domain of over-protecting the user.

What I propose is, why don't we submit one or another, and discuss it in more details?

File: torch/nn/quantized/modules/functional_modules.py (hunk 17 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/functional_modules.py]: I don't think it's a bug, as it is documented in the [Type Refinement](https://pytorch.org/docs/stable/jit.html#optional-type-refinement) section of the jit docs. @jamesr66a Is there a way to use optional args, while dropping the `if None` test?

File: torch/nn/quantized/modules/functional_modules.py (hunk 16 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/functional_modules.py]: Does that mean that QAT and PTQ observers have different semantics?

Comment (dzhulgakov) [reply torch/nn/quantized/modules/functional_modules.py]: That's the current observer semantics, if non-None is returned, it overrides the return: https://github.com/pytorch/pytorch/blob/77c08aa46c3f3460b95b89cbe357b99180bc824d/torch/nn/modules/module.py#L549-L551

File: torch/nn/quantized/modules/functional_modules.py (hunk 17 above)
Comment (dzhulgakov) [reply torch/nn/quantized/modules/functional_modules.py]: Actually, you're right, it's not a optional arg, it's an arg with default value: https://github.com/pytorch/pytorch/blob/77c08aa46c3f3460b95b89cbe357b99180bc824d/aten/src/ATen/native/native_functions.yaml#L399

Then why you just don't duplicate the signature to be the same, i.e. `int dim = 0`?

File: torch/nn/quantized/modules/functional_modules.py (hunk 19 above)
Comment (dzhulgakov) [reply torch/nn/quantized/modules/functional_modules.py]: I'll use @soumith as a tie-breaker - he voted for no name when we spoke earlier.

@raghuramank100 - in your example, it's still needed to align function types in two places (for example if one comments out some code)

File: torch/nn/quantized/modules/functional_modules.py (hunk 17 above)
Comment (jamesr66a) [reply torch/nn/quantized/modules/functional_modules.py]: I think the ideal solution in python would be to use kwargs and forward them directly:

```
import torch

dim = None

x = [torch.rand(3, 4) for i in range(3)]

def cat(x, **kwargs):
    torch.cat(x, **kwargs)

cat(x, dim=1)
cat(x)
```

However, this isn't supported by script, so what's used here is fine. @zafartahirov also proposed making the default `dim` equal to ` in the function signature here, which would probably also work

File: torch/nn/quantized/modules/functional_modules.py (hunk 16 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/functional_modules.py]: Got it! I will add a TODO, and can send a diff later on, while bringing @jerry39213gh into the loop

File: torch/nn/quantized/modules/functional_modules.py (hunk 19 above)
Comment (raghuramank100) [reply torch/nn/quantized/modules/functional_modules.py]: Yes, we still need to align function names in two places.  Lets go with removing the name as an argument. A todo will be to have torchscript ensure that observer instantiations are unique.

File: torch/nn/quantized/modules/functional_modules.py (hunk 16 above)
Comment (raghuramank100) [reply torch/nn/quantized/modules/functional_modules.py]: This should be a very simple diff: https://github.com/pytorch/pytorch/pull/24194. I just sent it out. Also, it should be @jerryzh168

**Review by raghuramank100 — APPROVED** [review]
